### DISCUSSION TITLE
Apply overload during cyclic loading

### DISCUSSION THREAD

**Discussion by Ming-is** [neperfepx/FEPX#83] [Q&A] [answered]
I try to use fepx software to analyze the deformation behavior of alloy under constant stress amplitude cyclic loading with overload. The yield stress of this material is above 800MPa. I want to apply 2 times of overload stress amplitude to it, but it fails. The error report shows that it does not converge or exceeds the hardening limit. So I would like to ask if fepx can do cyclic overload simulation? If I can, how can I achieve it?

The following is my congfig.

```
## Optional Input
 hard_type cyclic_isotropic
    load_tol_abs 10.0
    load_tol_rel 0.01

## Material Parameters

    number_of_phases 1

    phase 1

    crystal_type HCP
    m 0.010d0
    gammadot_0 1.0d0
    h_0 190.0d0
    g_0 390.0d0 468.0d0 663.0d0
    g_s0 530.0d0
   
    n 1.0d0
    c11 1.6966e5
    c12 0.8866e5
    c13 0.6166e5
    c44 0.4250e5
    c_over_a 1.587
    cyclic_a 0.050d0
    cyclic_c 3.50d0
  

## Deformation History
def_control_by uniaxial_load_target
   number_of_load_steps 14
    
    target_load 800 5 0.5 print_data
    target_load 0 5 0.5 print_data
    target_load -800 5 0.5 print_data
    target_load 0 5 0.5 print_data
    target_load 800 5 0.5 print_data
    target_load 1600 5 0.5 print_data
    target_load 800 5 0.5 print_data
     target_load 0 5 0.5 print_data
     target_load -800 5 0.5 print_data
    target_load -1600 5 0.5 print_data
    target_load -800 5 0.5 print_data
    target_load 0 5 0.5 print_data
    target_load 800 5 0.5 print_data
     target_load 0 5 0.5 print_data
     


    

## Boundary Condition

boundary_conditions uniaxial_grip
loading_direction Z
loading_face z0
strain_rate 0.0001
```

**Comment by mkasemer** [comment, marked as answer]
@Ming-is: The issue of poor convergence is likely that you want to take the jump to 1600 in too few steps. You are deforming, seemingly, well into the plastic regime, and you should take (at least initially) smaller steps so the solver knows the correct solution trajectory. Further, there is no guarantee given the material parameters that you have that you can reach a target load of 1600 - it could very well be that your material behavior saturates at a lower stress (and thus load), and you will not converge to the proper solution.

I suggest first running a monotonic test to understand the stress-strain behavior better, especially with regard to hardening, well into the plastic regime. Characterizing this behavior is important to know how your material parameters result in certain behavior.

**Reply by Ming-is** [reply]
The strain controlled monotonic tensile simulation I ran is shown in the figure. The yield strength exceeds 1200Mpa. I want to study the stress cyclic loading under overload. Although I have also tried to add loading steps, overload simulation still can not be realized. It seems that fepx cannot run overload simulation. I have tried ABAQUS to run overload simulation and use displacement to control loading, which is feasible in most cases. But fepx doesn't seem to have similar functions. I'm still confused whether fepx can simulate overload.

**Reply by rquey** [reply]
@Ming-is, you didn't attached any figure.

By *overload*, do you mean a stress greater than the yield stress, which brings the material well in the plastic regime, but smaller than the ultimate stress at failure?

If the answer is yes, then it can be simulated with a proper config file, and you should follows the procedure detailed by @mkasemer.

**Reply by Ming-is** [reply]
The missing figure is as follows:
![1_stress_strain](https://user-images.githubusercontent.com/71123582/167565198-fc27cfb8-3d85-44ad-97e1-eba171da17ab.png)
yes，Because some materials have the ability to withstand occasional overload without fracture, this overload may exceed twice or more of the yield limit. So I want to simulate this situation. I wonder if there is any suitable way to use fepx.

**Reply by rquey** [reply]
If you try to load this material monotonically to 1600 MPa, it will break, whether you do it experimentally or numerically.

What is an *occasional overload*? How do you apply it?

**Reply by Ming-is** [reply]
The overload is interspersed in the cyclic loading of equal stress amplitude, for example, the overload of plus or minus 1600mpa is loaded after every 10 cyclic loading of equal stress amplitude tension and compression of 1200Mpa, which is regarded as a large cycle to study the deformation behavior.

**Reply by rquey** [reply]
Okay.  So, you have to make sure that your crystal model and parameters (hardening, etc.) are so that the material can reach 1600 MPa. See the different [hardening models](https://fepx.info/doc/input.html#material-description) available in FEPX.

**Reply by Ming-is** [reply]
However, just adjusting the values of hardening and other parameters will not change the original yield strength or not conform to the actual cyclic hardening law. In fact, the constitutive parameters I used in this test are based on the constitutive parameters in Ti6Al4V the fepx 's example-2. I think this constitutive parameter is very accurate, and the adjustment may not be of great significance.
Moreover, I found that overload is often related to crack propagation, and the constitutive model used by fepx may not be able to simulate overload. I don't consider the simulation of crack propagation for the time being, but just try to analyze the impact of overload on other deformation behaviors, such as the start of slip system, so I'm still confused about whether the constitutive model currently used by fepx can simulate overload.

**Reply by mkasemer** [reply]
Overall I agree with @rquey 's assessment. If you believe that the above stress-strain curve accurately portrays the monotonic behavior, then I agree that a 1600MPa load is not realistically attainable considering ductile models. There is the possibility of different strain rates leading to different stress-strain behavior, but if you are considering Ti64, then this material exhibits little rate dependence and will not exhibit a significant jump in stress based on changes to the strain rate.

Further, you are now describing crack-propagation in the material as a possible driving factor. Firstly, no, FEPX (and indeed most 3D polycrystal plasticity frameworks) does not consider brittle fracture models. I also question whether this would lead to a dramatic increase in stress, 33% above the yield stress in a material that exhibits little hardening (nearly perfectly elastic-plastic). Either way, certainly if you are trying to model crack propagation, then FEPX is not the correct framework, and I have no suggestion as to what to pursue beyond implementation of crack propagation in FEPX (an undoubtedly non-trivial task).

**Reply by Ming-is** [reply]
I have no intention to analyze crack propagation. The above is just a shallow guess of my current situation. I think I see what you mean. In fact, fepx is still unable to realize overload cycle simulation

**Reply by mkasemer** [reply]
@Ming-is: to be clear, I don't think this is a limitation of FEPX. Crystal plasticity modeling, in general, will not result in the behavior you are describing, unless it is the result of a higher-order phenomenon that is currently not considered (such as crack propagation). I do not believe you will be able to attain the behavior you desire by using any CP framework (unless I have a severe misunderstanding of what you are attempting to do).
